packages/web3-eth-contract/src/contract.ts

```diff
@@ -895,57 +895,14 @@ export class Contract<Abi extends ContractAbi>
 					throw new Web3ValidatorError(errors);
 				}
 			}
-			if (methodAbi.stateMutability === 'payable') {
```

Comment:
What is the reason for removing 50 lines of code?

Reply:
Actually, the code was like this:
if (methodAbi.stateMutability === 'payable') {
    return methods
}
return the same methods as above except that some types are different, but there is a casing afterward in both cases. 

And for this, I removed the duplicates and consolidated the code for easier readability, maintenance, and testing :smile: